### PR TITLE
fix(shareFolder): #MAG-287 fix share a folder inside a folder

### PR DESCRIPTION
**Pull request by Rseuret**
## Describe your changes

fix share a folder inside a folder

## Issue ticket number and link

https://jira.support-ent.fr/browse/MAG-287

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/controller/ShareBoardController.java [hunk 1]
```diff
@@ -260,9 +260,9 @@ private Future<List<String>> getGroupUsers(UserInfos user, String id) {
     private void handleShareFolder(HttpServerRequest request, UserInfos user, List<SharedElem> newSharedElem, String id, I18nHelper i18nHelper, JsonObject share) {
 
         Future<List<SharedElem>> deletedRightFuture = this.magnetoShareService.getDeletedRights(id, newSharedElem, CollectionsConstant.FOLDER_COLLECTION);
-        this.getBookmarksToElems(user, newSharedElem).compose(bookmarkShared -> {
+        this.getBookmarksToElems(user, newSharedElem).onSuccess(bookmarkShared -> {
```
Comment (alicedraillard) [inline src/main/java/fr/cgi/magneto/controller/ShareBoardController.java:263]: retour à la ligne pour le .onSuccess

File: src/main/java/fr/cgi/magneto/controller/ShareBoardController.java [hunk 2]
```diff
@@ -260,9 +260,9 @@ private Future<List<String>> getGroupUsers(UserInfos user, String id) {
...
             newSharedElem.addAll(bookmarkShared);
-            return this.magnetoShareService.checkParentRights(id, newSharedElem, CollectionsConstant.FOLDER_COLLECTION)
+            this.magnetoShareService.checkParentRights(id, newSharedElem, CollectionsConstant.FOLDER_COLLECTION)
```
Comment (alicedraillard) [inline src/main/java/fr/cgi/magneto/controller/ShareBoardController.java:265]: pourquoi enlever le return ?

Comment (Rseuret) [reply src/main/java/fr/cgi/magneto/controller/ShareBoardController.java:265]: je sais pas pourquoi mais de temprs en temps ca attendait pas la fin de la premiere future

File: src/main/java/fr/cgi/magneto/controller/ShareBoardController.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
 import java.util.stream.Collectors;
 
 /**
- *Handle folders too but can't rename without losing old shared Boards
+ * Handle folders too but can't rename without losing old shared Boards
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/controller/ShareBoardController.java:31]: indetntation

File: src/main/java/fr/cgi/magneto/controller/ShareBoardController.java [hunk 4]
```diff
@@ -242,7 +242,7 @@ private SharedElem createElemFromNeo(SharedElem elem, String id, String type) {
     private Future<List<String>> handlePromiseWithCompositeFuture(Promise<List<String>> promise, List<Future<List<String>>> futures, List<String> usersIdToShare) {
         if (!futures.isEmpty()) {
             FutureHelper.all(futures).onSuccess(
-                        result -> promise.complete(result.list().stream().flatMap(elem -> ((List<String>) elem).stream()).collect(Collectors.toList()))
+                    result -> promise.complete(result.list().stream().flatMap(elem -> ((List<String>) elem).stream()).collect(Collectors.toList()))
```
Comment (Rseuret) [inline src/main/java/fr/cgi/magneto/controller/ShareBoardController.java:245]: indetntation

**Review by flmartineau — APPROVED** [review]

**Pull request merged**